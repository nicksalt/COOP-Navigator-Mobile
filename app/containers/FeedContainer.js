import React from "react";
import Feed from "../components/Feed";

export default class FeedContainer extends React.Component {
  state = {
    notifications: temp
  };

  render() {
    return <Feed notifications={this.state.notifications} />;
  }
}

const temp = [
  {
    key: 0,
    icon: "md-calendar",
    title: "Co-op Key Date",
    date: "Oct 18, 2017",
    description:
      "Last day to update your resume for the Winter Coop term is Oct 30, 2017",
    footerText: "Fall - 2nd, 3rd, or 4th work term only"
  },
  {
    key: 1,
    icon: "ios-people",
    title: "New interview!",
    date: "Feb 1, 2018",
    description:
      "IBM selected you for an interview for the position of Junior Software developer",
    footerText: "Winter work term"
  },
  {
    key: 2,
    icon: "ios-trophy",
    title: "Congratulations!",
    date: "Feb 10, 2018",
    description:
      "Google really liked you when you interviewed with them. They have offered you a position as a Software Engineer on their Android team!",
    footerText: "Summer Work Term"
  },
  {
    key: 3,
    icon: "md-calendar",
    title: "Co-op Key Date",
    date: "Apr 27, 2018",
    description:
      "Please head on over to the navigator and confirm your intention to participate in coop before ",
    footerText: "Fall - All work terms"
  },
  {
    key: 4,
    icon: "ios-people",
    title: "New interview!",
    date: "Feb 1, 2018",
    description:
      "YouTube selected you for an interview for the position of Junior Software developer",
    footerText: "Winter work term"
  },
  {
    key: 5,
    icon: "ios-people",
    title: "New interview!",
    date: "Feb 1, 2018",
    description:
      "YouTube selected you for an interview for the position of Junior Software developer",
    footerText: "Winter work term"
  }
];
