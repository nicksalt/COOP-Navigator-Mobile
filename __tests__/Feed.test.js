import React from "react";
import FeedContainer from "../app/containers/FeedContainer";
import renderer from "react-test-renderer";

jest.mock("FlatList", () => "FlatList");

test("Feed snapshot renders correctly", () => {
  const tree = renderer.create(<FeedContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});
