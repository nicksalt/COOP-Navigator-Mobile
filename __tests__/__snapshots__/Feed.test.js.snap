// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed snapshot renders correctly 1`] = `
<FlatList
  data={
    Array [
      Object {
        "date": "Oct 18, 2017",
        "description": "Last day to update your resume for the Winter Coop term is Oct 30, 2017",
        "footerText": "Fall - 2nd, 3rd, or 4th work term only",
        "icon": "md-calendar",
        "key": 0,
        "title": "Co-op Key Date",
      },
      Object {
        "date": "Feb 1, 2018",
        "description": "IBM selected you for an interview for the position of Junior Software developer",
        "footerText": "Winter work term",
        "icon": "ios-people",
        "key": 1,
        "title": "New interview!",
      },
      Object {
        "date": "Feb 10, 2018",
        "description": "Google really liked you when you interviewed with them. They have offered you a position as a Software Engineer on their Android team!",
        "footerText": "Summer Work Term",
        "icon": "ios-trophy",
        "key": 2,
        "title": "Congratulations!",
      },
      Object {
        "date": "Apr 27, 2018",
        "description": "Please head on over to the navigator and confirm your intention to participate in coop before ",
        "footerText": "Fall - All work terms",
        "icon": "md-calendar",
        "key": 3,
        "title": "Co-op Key Date",
      },
      Object {
        "date": "Feb 1, 2018",
        "description": "YouTube selected you for an interview for the position of Junior Software developer",
        "footerText": "Winter work term",
        "icon": "ios-people",
        "key": 4,
        "title": "New interview!",
      },
      Object {
        "date": "Feb 1, 2018",
        "description": "YouTube selected you for an interview for the position of Junior Software developer",
        "footerText": "Winter work term",
        "icon": "ios-people",
        "key": 5,
        "title": "New interview!",
      },
    ]
  }
  keyExtractor={[Function]}
  renderItem={[Function]}
/>
`;
